ecocheck.py

```diff
@@ -0,0 +1,118 @@
+import logging
+from infra.system.helpers.utils import getScriptDir
+from infra.system.helpers.schema_validator import SchemaValidator
+import time
+import datetime
+from gofed.lib.pkgdb2client import get_packages
+from gofed.lib.bitbucketclient import get_commits_bitbucket
+from gofed.lib.githubclient import get_commits_github
+from gofed.lib.kojiclient import KojiClient
+
+
+SUPPORTED_REPOSIRORIES = ['github.com', 'bitbucket.org']
+
+
+def repository_info(clone_url):
+	if not clone_url:
+		return []
+	data = clone_url.split('/')
+	if data[2] in SUPPORTED_REPOSIRORIES:
+		repository = {
+			'host': data[2],
+			'owner': data[3],
+			'package': data[4].split('.')[0]
+		}
+	else:
+		logging.error('Unsupported upstream host: %s' % data[2])
+		return []
+	return repository
+
+
+class EcoCheck:
+	def __init__(self):
```

Comment:
Builder would be suitable for this class. Builder can provide more ways how to initialize the class (inject fake or various real clients). E.g.

```
def buildFullEcoCheck(self):
    koji_client = KojiClient()
    github_client = GithubClient()
    bitbucket_client = BitbucketClient()
    pkgdb_client = PkgdbClient()
    return EcoCheck(koji_client, github_client, bitbucket_client, pkgdb_client)

def buildFakeEcoCheck(self):
    koji_client = FakeKojiClient()
    github_client = FakeGithubClient()
    bitbucket_client = FakeBitbucketClient()
    pkgdb_client = FakePkgdbClient()
    return EcoCheck(koji_client, github_client, bitbucket_client, pkgdb_client)
```